.github/workflows/ci.yml

```diff
@@ -0,0 +1,105 @@
+---
+name: CI
+on: [push]
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.41
+        env:
+          VERBOSE: "true"
+
+  tidy:
+    name: Tidy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Check if mods are tidy
+        run: make check-tidy
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Run tests
+        run: make test
+        env:
+          VERBOSE: "true"
+
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    needs: [test]
```

Comment:
I'm cautious of making build dependent on tests. You might want the binaries/docker image to play with before writing tests.

Reply:
True, but it's each enough to do so locally. My thinking was mostly that we don't want to build them if there are failing tests cause they might break shit if someone actually does try using them, hence let the dev do it locally on their machine instead. `make build` will give a binary that works locally, `make build-snapshot` will do the same as the `build` job here.